dashboard/app/models/user.rb

```diff
@@ -800,6 +800,38 @@ def User.track_script_progress(user_id, script_id)
     end
   end
 
+  # If appropriate, increases the level counts for the concept-difficulties
+  # associated with the completed level.
+  def User.track_proficiency(user_id, script_id, level_id)
+    level_concept_difficulty = LevelConceptDifficulty.where(level_id: level_id).first
+    if !level_concept_difficulty
+      return
+    end
+
+    # Exit early if a hint was used, as no proficiency is gained in this case.
```

Comment:
It seems a little bit odd that the calling function is responsible for checking if perfection has been achieved, but this is responsible for making sure it's REALLY been achieved. What do you think about rolling this check into `new_level_perfected`?